integration/firestore/firebase_export.ts

```diff
@@ -0,0 +1,56 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as firebase from 'firebase';
+
+// This file replaces "packages/firestore/test/integration/util/firebase_export"
+// and depends on the minified sources.
+
+let appCount = 0;
+
+export function newTestFirestore(
+  projectId: string,
+  nameOrApp?: string | firebase.app.App,
+  settings?: firebase.firestore.Settings
+): firebase.firestore.Firestore {
+  if (nameOrApp === undefined) {
+    nameOrApp = 'test-app-' + appCount++;
+  }
+  const app =
+    typeof nameOrApp === 'string'
+      ? firebase.initializeApp({ apiKey: 'fake-api-key', projectId }, nameOrApp)
+      : nameOrApp;
+
+  const firestore = firebase.firestore(app);
+  if (settings) {
+    firestore.settings(settings);
+  }
+  return firestore;
+}
+
+export function usesModularApi(): false {
```

Comment:
"Modular" doesn't really cleanly describe the difference between the the Firestore-API-with-classes vs the tree-shakeable one.

The core of the difference seems to be class-based vs function-based. To me this suggests that this should be the basis of this kind of property.